### PR TITLE
feat: SPFx add web part handling version

### PR DESCRIPTION
**Pull request by yuqizhou77**
[Feature 26007373](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/26007373): SPFx add web part by defaut using latest version which bring version mismatch to the original project. 

When adding web part, we compare solution version, version of generator installed globally by the user and the version of generator installed locally by TTK, and try to select the package which is in the same version with solution version to continue. If we cannot find, then we will confirm with users to install/upgrade/continue with inconsistent version.

- Confirm with user if need to install local dependencies
image

- Confirm with user if need to upgrade local dependencies
image

- Confirm with user to continue 
image

TODO:
- help doc update

### CODE REVIEW

**Review by nliu-ms — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -111,6 +111,17 @@
...
+  "plugins.spfx.addWebPart.versionMismatch.help": "Help",
+  "plugins.spfx.addWebPart.versionMismatch.continue": "Continue",
+  "plugins.spfx.addWebPart.versionMismatch.output": "Your SPFx solution version is %s. You have %s installed globally and %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
+  "plugins.spfx.addWebPart.versionMismatch.localOnly.output": "Your SPFx solution version is %s. You have %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
+  "plugins.spfx.addWebPart.cannotFindSolutionVersion": "Cannot find the version of your SPFx solution in %s",
+  "plugins.spfx.error.installDependencyError": "Encountered unknown issues when setting up SPFx environment in %s folder. You could follow %s to install %s to set up global SPFx environment.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:124]: You could follow -> Follow

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -111,6 +111,17 @@
...
+  "plugins.spfx.addWebPart.versionMismatch.help": "Help",
+  "plugins.spfx.addWebPart.versionMismatch.continue": "Continue",
+  "plugins.spfx.addWebPart.versionMismatch.output": "Your SPFx solution version is %s. You have %s installed globally and %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
+  "plugins.spfx.addWebPart.versionMismatch.localOnly.output": "Your SPFx solution version is %s. You have %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
+  "plugins.spfx.addWebPart.cannotFindSolutionVersion": "Cannot find the version of your SPFx solution in %s",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:123]: Unable to find SPFx version in your solution in %s

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -111,6 +111,17 @@
...
+  "plugins.spfx.addWebPart.install": "Install",
+  "plugins.spfx.addWebPart.confirmUpgrade": "Teams Toolkit is using SPFx %s, but detected your SPFx solution is %s. Would you like to upgrade SPFx version in Teams Toolkit directory to %s and add web part?",
+  "plugins.spfx.addWebPart.upgrade": "Upgrade",
+  "plugins.spfx.addWebPart.versionMismatch.continueConfirm": "Teams Toolkit detected your SPFx solution is %s, but we cannot find this version in your machine. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). The version mismatch may cause unexpected error. Would you like to continue?",
+  "plugins.spfx.addWebPart.versionMismatch.help": "Help",
+  "plugins.spfx.addWebPart.versionMismatch.continue": "Continue",
+  "plugins.spfx.addWebPart.versionMismatch.output": "Your SPFx solution version is %s. You have %s installed globally and %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:121]: Your SPFx solution version is %s. -> SPFx version in your solution is %s.
can delete "If continue to add web part,", the version -> The version

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -111,6 +111,17 @@
...
+  "plugins.spfx.addWebPart.versionMismatch.localOnly.output": "Your SPFx solution version is %s. You have %s installed in Teams Toolkit directory. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). If continue to add web part, the version mismatch may cause unexpected error. Find possible solutions in %s.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:122]: Your SPFx solution version is %s.  -> SPFx version in your solution is %s.
can delete "If continue to add web part, "
the version -> The version

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -111,6 +111,17 @@
...
   "plugins.spfx.import.log.success": "Teams Toolkit has imported your SPFx solution successfully. A complete log of import details can be found in %s.",
   "plugins.spfx.import.log.fail": "Teams Toolkit failed to import your SPFx solution. A complete log of import details can be found in %s.",
+  "plugins.spfx.addWebPart.confirmInstall": "Teams Toolkit detected your SPFx solution is %s, but you don't have this version installed on your machine. Would you like to install SPFx %s in Teams Toolkit directory to continue adding web part?",
+  "plugins.spfx.addWebPart.install": "Install",
+  "plugins.spfx.addWebPart.confirmUpgrade": "Teams Toolkit is using SPFx %s, but detected your SPFx solution is %s. Would you like to upgrade SPFx version in Teams Toolkit directory to %s and add web part?",
+  "plugins.spfx.addWebPart.upgrade": "Upgrade",
+  "plugins.spfx.addWebPart.versionMismatch.continueConfirm": "Teams Toolkit detected your SPFx solution is %s, but we cannot find this version in your machine. Teams Toolkit uses SPFx installed in Teams Toolkit directory by default (%s). The version mismatch may cause unexpected error. Would you like to continue?",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:118]: Teams Toolkit detected your SPFx solution is %s, but we cannot find this version in your machine. -> SPFx version in your solution is %s, not yet installed on this machine.

Would you like -> Do you want

File: packages/fx-core/resource/package.nls.json [hunk 6]
```diff
@@ -111,6 +111,17 @@
   "plugins.spfx.import.success": "Your SPFx solution has been successfully imported to %s.",
   "plugins.spfx.import.log.success": "Teams Toolkit has imported your SPFx solution successfully. A complete log of import details can be found in %s.",
   "plugins.spfx.import.log.fail": "Teams Toolkit failed to import your SPFx solution. A complete log of import details can be found in %s.",
+  "plugins.spfx.addWebPart.confirmInstall": "Teams Toolkit detected your SPFx solution is %s, but you don't have this version installed on your machine. Would you like to install SPFx %s in Teams Toolkit directory to continue adding web part?",
+  "plugins.spfx.addWebPart.install": "Install",
+  "plugins.spfx.addWebPart.confirmUpgrade": "Teams Toolkit is using SPFx %s, but detected your SPFx solution is %s. Would you like to upgrade SPFx version in Teams Toolkit directory to %s and add web part?",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:116]: Teams Toolkit is using SPFx version %s, but SPFx version in your solution is %s. Do you want to upgrade SPFx version in Teams Toolkit directory to %s and add web part?

File: packages/fx-core/resource/package.nls.json [hunk 7]
```diff
@@ -111,6 +111,17 @@
   "plugins.spfx.import.success": "Your SPFx solution has been successfully imported to %s.",
   "plugins.spfx.import.log.success": "Teams Toolkit has imported your SPFx solution successfully. A complete log of import details can be found in %s.",
   "plugins.spfx.import.log.fail": "Teams Toolkit failed to import your SPFx solution. A complete log of import details can be found in %s.",
+  "plugins.spfx.addWebPart.confirmInstall": "Teams Toolkit detected your SPFx solution is %s, but you don't have this version installed on your machine. Would you like to install SPFx %s in Teams Toolkit directory to continue adding web part?",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:114]: SPFx version in your solution is %s, not yet installed on your machine. Do you want to install SPFx %s in Teams Toolkit directory to continue adding web part?

**Review by sffamily — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**